### ISSUE TITLE
Deprecation: Detect missing explicit access check on entity queries

### ISSUE THREAD

**Issue by mxr576** [mglaman/phpstan-drupal#315]
See: https://www.drupal.org/node/3201242

If I am not mistaken those are currently not detected by the following versions:
```
mglaman/drupal-check                                      1.2.1         CLI tool for running checks on a Drupal code base
mglaman/phpstan-drupal                                    1.1.10        Drupal extension and rules for PHPStan
phpstan/phpstan                                           1.4.3         PHPStan - PHP Static Analysis Tool
phpstan/phpstan-deprecation-rules                         1.0.0         PHPStan rules for detecting usage of deprecated classes, methods, properties, constants and traits.
```

**Comment by mxr576** [comment]
Related: https://www.drupal.org/project/rector/issues/3261475

**Comment by mglaman** [comment]
Wow, that sucks. It's a runtime check. 

That's going to be nearly impossible for Rector. Luckily some recent work here makes it possible.

**Comment by mxr576** [comment]
I hoped so :) :crossed_fingers: 

Not sure about the impossibility with Rector, but that is a long shot generic solution idea.
